lib/reply.js

```diff
@@ -138,9 +138,24 @@ Reply.prototype.send = function (payload) {
 
   if (this[kReplySerializer] !== null) {
     payload = this[kReplySerializer](payload)
-  } else if (hasContentType === false || contentType.indexOf('application/json') > -1) {
-    if (hasContentType === false || contentType.indexOf('charset') === -1) {
+
+  // We test below if we have a custom JSON type (e.g. 'application/hal+json' or 'application/ld+json')
+  } else if (hasContentType === false || /^application\/([a-z.-]+\+)?json/.test(contentType)) {
+    if (hasContentType === false) {
       this[kReplyHeaders]['content-type'] = CONTENT_TYPE.JSON
+    } else {
+      // If hasContentType === true, we have a JSON mimetype
+      if (contentType.indexOf('charset') === -1) {
+        // We extract the custom mimetype part (e.g. 'hal+' from 'application/hal+json')
+        const customJsonType = this[kReplyHeaders]['content-type'].substring(
+          this[kReplyHeaders]['content-type'].indexOf('/'),
```

Comment:
I wonder if cost of accessing `this[kReplyHeaders]['content-type']` multiple times is smaller than cost of introducing interim value to keep it there.

Also maybe there should be a short-circuit for most common path of having exact match of `application/json`?

Reply:
I agree. I moved it to a variable.